### ISSUE TITLE
ERROR: could not locate path that contains java

### ISSUE THREAD

**Issue by TrinityMHT** [itzg/docker-minecraft-server#2086]
### Describe the problem

Hi.

I'm trying to run a server on Java 19 but I'm getting an error.
All other versions of Java 8, 11, 17 work fine.

### Container definition

```
Paste run command or compose file here
```

### Container logs

```
[init] ERROR: your Docker provider has an annoying flaw where it
[init] tries to set PATH even though the container establishes
[init] a very specific value.
[init] ERROR: could not locate path that contains java
```

**Comment by itzg** [comment]
What is your Docker provider?

**Comment by TrinityMHT** [comment]
```
root@server:~# uname -a
Linux server 4.9.0-15-amd64 #1 SMP Debian 4.9.258-1 (2021-03-08) x86_64 GNU/Linux
root@server:~# cat /etc/issue
Debian GNU/Linux 9 \n \l
```

**Comment by itzg** [comment]
How do you run Docker?

**Comment by TrinityMHT** [comment]
`docker container run -d -it -e UID=$(id -u s1) -e GID=$(id -u s1) -p 25565:25565 -v /home/servers/s1:/data -e TZ=Europe/Warsaw -e EULA=TRUE -e GUI=FALSE -e CUSTOM_SERVER=server.jar --name s1 itzg/minecraft-server:java19`

**Comment by TrinityMHT** [comment]
The problem only occurs on Java 19.

**Comment by itzg** [comment]
Thanks for the additional context. I was suspecting a Portainer issue, but as you have found might be something about the java8 base image.

**Comment by TrinityMHT** [comment]
I don't use Portainer. Linux command line only. Do you know any solution that would help me get my server running on java 19?

**Comment by itzg** [comment]
Why do you need Java 19?

Otherwise, I am not able to look into this soon, so you will need to further investigate and a PR would be great.

**Comment by rhullah** [comment]
What is the $PATH set to when it runs? You can test this in the container by tweaking the run command:

```shell
docker container run --rm -it -e UID=$(id -u s1) -e GID=$(id -u s1) -p 25565:25565 -v /home/servers/s1:/data -e TZ=Europe/Warsaw -e EULA=TRUE -e GUI=FALSE -e CUSTOM_SERVER=server.jar --name s1 --entrypoint /usr/bin/bash itzg/minecraft-server:java19
```

Once running you'll be at a bash prompt in the container.  Try some of the following command to get an idea of the state of the container:

```shell
echo $PATH
which java
command -v java
```

You can then exit the container by type `exit`

**Comment by TrinityMHT** [comment]
```
root@dd6d64b630ae:/data# echo $PATH
/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
root@dd6d64b630ae:/data# which java
root@dd6d64b630ae:/data# command -v java
/opt/java/openjdk/bin/java
root@dd6d64b630ae:/data#
```

**Comment by rhullah** [comment]
😕 It's confusing that `which` failed but `command` worked.  Instead of `command java --version` you should instead try just `java --version`  if that doesn't work (which it should), then try `/opt/java/openjdk/bin/java --version`

**Comment by rhullah** [comment]
Oh, that `command java --version` should have worked (sorry about that).  That is odd.  Are you sure you have the latest image, maybe re-pull the image?

**Comment by rhullah** [comment]
Reading through multiple issues on the error generated it seems that it could either be the base adoptium image the container is working with, or the version of docker installed.  Are you running the latest docker version?

**Comment by TrinityMHT** [comment]
```
root@7029f805e9e6:/data# command java --version
[0.007s][warning][os,thread] Failed to start thread "GC Thread#0" - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.
[0.008s][error  ][gc,task  ] Failed to create worker thread
root@7029f805e9e6:/data# /opt/java/openjdk/bin/java --version
[0.009s][warning][os,thread] Failed to start thread "GC Thread#0" - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.
[0.009s][error  ][gc,task  ] Failed to create worker thread
root@7029f805e9e6:/data# java --version
[0.008s][warning][os,thread] Failed to start thread "GC Thread#0" - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.
[0.011s][error  ][gc,task  ] Failed to create worker thread
root@7029f805e9e6:/data# command java --version
[0.009s][warning][os,thread] Failed to start thread "GC Thread#0" - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.
[0.009s][error  ][gc,task  ] Failed to create worker thread
root@7029f805e9e6:/data#

```

```
root@server:~# docker version
Client: Docker Engine - Community
 Version:           19.03.9
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        9d988398e7
 Built:             Fri May 15 00:24:16 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.9
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       9d988398e7
  Built:            Fri May 15 00:22:48 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.2
  GitCommit:        9754871865f7fe2f4e74d43e2fc7ccd237edcbce
 runc:
  Version:          1.0.0-rc6+dev
  GitCommit:        09c8266bf2fcf9519a651b04ae54c967b9ab86ec
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Comment by rhullah** [comment]
It looks like you're on an older Debian version (v9).  It looks like the latest Docker supports v10 and newer of Debian.  Do you know if you're able to update Docker to any newer version or if you're at the max of what's supported for your Debian version?

**Comment by rhullah** [comment]
Something else you could try if you wanted to test something before a Docker/OS upgrade is if different images experience the same thing.  `itzg/minecraft-server:java19` is based off `eclipse-temurin:19-jre-jammy` (if I'm following the code correctly).  You could try each one and see if you get a working java executable, `java --version`, in each of them substituting in the appropriate `image:tag`:

`docker container run --rm -it image:tag`

Using the following tags:
- `eclipse-temurin:19-jre-jammy`
- `eclipse-temurin:19-jre-focal`
- `eclipse-temurin:19-jre-alpine`

**Comment by TrinityMHT** [comment]
After testing, it turns out that only 19-jre-jammy does not work.

19-jre-focal and 19-jre-alpine work fine.

**Comment by rhullah** [comment]
Yeah, I figured that was going to happen becaus `itzg/minecraft-server:java17` worked for you and that was based off of `focal`.  I think there's two solutions to this and that would be either upgrade your OS (at a minimum to v10) so that you can upgrade to the current Docker version (I looked and it seems you're on the latest/last version available for Debian v9), or change the base image for v19 to be focal based.  I'll leave that to be a discussion w/ you and @itzg. 😄

**Comment by TrinityMHT** [comment]
> Yeah, I figured that was going to happen because `itzg/minecraft-server:java17` worked for you and that was based off of `focal`. I think there's two solutions to this and that would be either upgrade your OS (at a minimum to v10) so that you can upgrade to the current Docker version (I looked and it seems you're on the latest/last version available for Debian v9), or change the base image for v19 to be focal based. I'll leave that to be a discussion w/ you and @itzg. 😄

Thank you very much for help.

@itzg Why is there focal in all Java versions and jammy in Java 19? Is it possible to set focal?

**Comment by itzg** [comment]
Meanwhile, found [the issue](https://github.com/itzg/docker-minecraft-server/issues/1898) that brought in Java 19 image and doesn't look like there was a requirement for jammy. Bumping down to focal makes the most sense all around.
